### ISSUE TITLE
GPX database upgrade crash

### ISSUE THREAD

**Issue by jfburke3** [osmandapp/Osmand#4125]
I e-mailed in the exception log from the problem phone a little while ago.

What version of OsmAnd are you using, on what device and Android/iOS version?
Nightly build 2.7.0 #22070M (but also happens on a previous build), on Android 4.2.2, Samsung Galaxy S4.

Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?
Offline, 

Both my own map built with OsmAndMapCreator, as well as your offline maps.  Both for US Georgia (mine, built a couple of days ago, but I also tried replacing it with your June 30 map).

If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs, and its edition date.

US Georgia, June 30, and mine from a few days ago.

What I observed:

First, important to note that the S4 I use for navigation doesn't have cellular service.  It's my old phone that I kept after upgrading to a new phone because it works perfectly fine for OSM navigation thanks to offline maps.  As such, if I'm not in range of my (or open) wifi, it doesn't have a network connection, which I suspect might be part of the problem.

Second, important to note that I was able to successfully open OsmAnd with my map file in place at least once after copying it to the phone.  AND, I have intermittently been able to get it to work on the S4.  I do not have any problems with the same map file on my Galaxy S7/Android 6.0.1 using nightly builds.

Saturday, I got in the car to go somewhere, put my S4 in the mount for routing (not that I needed it for that, but I use OsmAnd to make notes about improvements OSM needs).  Brought up OsmAnd, and instead of the usual displayed map, the map area was blank except for my location and Mapillary tracks.  After a short time, the screen changed from day to night coloring, still without the map.  Tried to open My Places, and it promptly crashed.  Relaunched, tried again, crashed again.  A little trial & error showed that menu items related to places on the map would crash it, but others (Settings, Plugins, etc.) would not.  Rebooting phone didn't help.

Decided to do my trip without navigation, but started transferring a fresh copy of _my_ Georgia map (the same one that I had on the S4) from my S7 to S4 while I was driving, in case of corrupt file.

After transfer finished, moved it to the OsmAnd directory, rebooted, relaunched OsmAnd, the above crash behavior persisted.  I still had the July 5 night build apk on the phone, so tried installing it since it worked for me earlier, but still crashed.  Transferred the osmandToInstall.apk from my S7 to S4 and installed it (it was a day newer than the original version I had on the S4.  I tend to have one on a newer build than the other.).  Still crashed.

When I was getting ready to leave my destination to go back home, thought that the problem might be related to network connection.  *Usually* when I leave home, I set my destination & other settings before I get in the car.  Also, started enabling the new Mapillary plugin and Offline Maps plugin (which I didn't usually have enabled on the S4).  So I disabled both of those plugins, and was then able to relaunch OsmAnd and have it function normally.  Yay.  Reenabled Mapillary plugin, it kept working.  Went home happy.

Once home, reenabled Offline Maps plugin, exited OsmAnd, disabled wifi on the S4.  Not sure if I killed the OsmAnd process or rebooted at this point.  Launched OsmAnd and it worked fine.  Hmph.

Sunday I wanted to go offroading, set a destination in OsmAnd for the beginning of the offroad trail I was going to drive (inside my apartment, with Mapillary and Offline Maps enabled, using build 22070 and my US Georgia map).  No problem.

Got in Jeep, drove to the trail start.  OsmAnd performed normally.  In OsmAnd selected new destination where I wanted to return to paved roads along with a couple of intermediate waypoints to force OsmAnd to route me along the trail and not paved roads to the destination.  It worked great for several hours, including deviations from the route to explore other trails.  Keep in mind this phone had no network connection the whole time.

When I reached the "return to paved roads" destination, I wanted to go home, so tried to select my saved "home" favorite.  OsmAnd crashed.  Tried a few times, killed the OsmAnd process, rebooted phone, it still wouldn't display the map, would switch to night mode, and still crashed when I tried to Search or open My Places, etc.  Tried disabling the Mapillary and Offline Maps plugins, no help.  Deleted ind.cache, no help.  Gave up, drove home without it (I knew the way, I just like running OsmAnd to help find OSM issues that need fixing, because it's a GREAT tool for that!).

At home, the same behavior continued.  Can't get OsmAnd to display the map and do anything map related without crashing.  (Side note:  when I do try to do a menu function, it *does* briefly display the map before opening that menu function or crashing (depending on the function)).  This is when I tried replacing my map file with yours, and rebooting again.  It still crashes.

**Comment by vshcherb** [comment]
Do you have a crash log you could attach to the issue?

**Comment by jfburke3** [comment]
I e-mailed one in, but here it is again.

[exception.txt](https://github.com/osmandapp/Osmand/files/1159695/exception.txt)

I can't find a specific set of events or circumstances that (1) cause the problem, and (2) resolve it temporarily.  Yesterday, somehow, I was able to get the app to start working properly again, and then later in the day, it returned to the "don't display the map and crash when I try certain menu functions" behavior.

**Comment by jfburke3** [comment]
Ref #4135 .  Sounds like the same thing.

**Comment by jfburke3** [comment]
I cannot reproduce the problem on my S7 / Android 6.0.1.  It only happens on the S4 / Android 4.2.2.

**Comment by sonora** [comment]
Unfortunately it is not fixed, I attach the crash log from the latest build.  [exception.log.txt](https://github.com/osmandapp/Osmand/files/1163938/exception.log.txt)


Side note: I have noticed for while now that all crashes seem to be contained in our exception.log files exactly twice (for the exact same time)?

**Comment by jfburke3** [comment]
I was about to report the same thing. The map displays now, though.

**Comment by vshcherb** [comment]
But now the situation is quite inconsistent. So unless you downgrade somehow to 2 months builds & upgrade to today build without intermediate build only then you can verify if the issue is fixed.

Right now your db is marked as upgraded, though it is not actually upgraded and it will keep crashing.

**Comment by vshcherb** [comment]
Today we are releasing 2.7.0 -beta let's see if get similar reports.

**Comment by jfburke3** [comment]
@vshcherb Yeah, that worked.  Thanks for the info.  I made a copy of my entire osmand directory, uninstalled the build I had, uninstalled OsmAnd, then downloaded OsmAnd-master-nb-2017-05-04.apk and installed that.  Copied my map files and favorites back to the osmand directory, then downloaded and installed OsmAnd-master-nb-2017-07-21.apk.  Everything seems to work.  Just need to finish copying my tracks, av notes, etc., back to the osmand directory.

I also installed the just-published beta on my S5 (yes, I have 3 phones) running Android 5.0, and it seems to be working just fine.  If you want me to try it on my S4, just let me know.
